partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/PlanTyper.kt

```diff
@@ -1225,26 +1163,30 @@ internal class PlanTyper(
                 isClosed && isOrdered -> {
                     struct.fields.firstOrNull { entry -> binding.isEquivalentTo(entry.key) }?.let {
                         (sensitive(it.key) to it.value)
-                    } ?: (key to MISSING)
+                    } ?: run {
+                        return null
+                    }
```

Comment:
think you don't need this `run`?

```suggestion
                    } ?: return null
```